src/kv/rocksdb_cache/ShardedCache.h

```diff
@@ -56,34 +56,33 @@ class ShardedCache : public rocksdb::Cache, public PriorityCache::PriCache {
  public:
   ShardedCache(size_t capacity, int num_shard_bits, bool strict_capacity_limit);
   virtual ~ShardedCache() = default;
+  // rocksdb::Cache
   virtual const char* Name() const override = 0;
```

Comment:
this has been removed here: https://github.com/ceph/rocksdb/compare/c540de6f709b66efd41436694f72d6f7986a325b...6a2bb9473340846d75d32aa8f3eac91c744f92d8#diff-ea5497937d78177864cccc84fd22d12fb94f4c18d000774a722d3457657733acL60
do we still need it?

Reply:
i am not able to find "this" by following the link. could you be more specific?

Reply:
@tchaikov here: https://github.com/facebook/rocksdb/pull/8297/files#diff-ea5497937d78177864cccc84fd22d12fb94f4c18d000774a722d3457657733acL76

Reply:
@ideepika you are looking at the wrong file / class. `ShardedCache` inherits from `rocksdb::Cache` not `rocksdb::ShardedCache`. also, if it's already removed, it'd fail the build as this virtual function is marked with `override`. if the parent class does not have this method, the compiler would refuse to compile this header file.

Reply:
but as the next step, i think we should inherit from `rocksdb::ShardedCache`, and drop the shared bits from this class, which will make it a customized `ShardedCache` which inherits from ``rocksdb::ShardedCache` and `PriorityCache::PriCache`.

Reply:
> but as the next step, i think we should inherit from rocksdb::ShardedCache, and drop the shared bits from this class, > > which will make it a customized ShardedCache which inherits from ``rocksdb::ShardedCacheandPriorityCache::PriCache`.

I agree, that should be ideal, should I add it to trello?

Reply:
not sure, i am still expecting @markhpc 's review. i suspect that he had good reason not to do so when implementing `BinnedLRUCache`.

Reply:
@kefu Sorry, I haven't had time to review yet.  To be honest it's been a while since I've thought about this code, but the primary thing we want to get in at some point (hopefully before quincy) is the work here:

https://github.com/markhpc/ceph/commit/0a318600ec0d187ea66afb9847c4e94e6d96b627

Now that we have some of the other changes layered on top of the column family sharding it's time to revisit it again.